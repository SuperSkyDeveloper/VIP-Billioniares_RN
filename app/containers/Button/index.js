import React from 'react';
import PropTypes from 'prop-types';
import {
  StyleSheet,
  View,
  Text,
  Image,
  ActivityIndicator,
  TouchableHighlight,
} from 'react-native';
import { RectButton } from 'react-native-gesture-handler';
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
import LinearGradient from 'react-native-linear-gradient';

import {
  COLOR_BUTTON_PRIMARY,
  COLOR_BUTTON_SECONDARY,
  COLOR_BUTTON_DEFAULT,
  COLOR_BUTTON_DANGER,
  COLOR_BUTTON_WHITE,
  COLOR_BUTTON_GRAY,
  COLOR_BUTTON_DONE,
  COLOR_BUTTON_FACEBOOK,
  COLOR_BUTTON_GOOGLE,
  COLOR_BUTTON_TEXT_PRIMARY,
  COLOR_BUTTON_TEXT_SECONDARY,
  COLOR_BUTTON_TEXT_DEFAULT,
  COLOR_BUTTON_TEXT_DANGER,
  COLOR_BUTTON_TEXT_WHITE,
  COLOR_BUTTON_TEXT_DONE,
  COLOR_BUTTON_TEXT_FACEBOOK,
  COLOR_BUTTON_TEXT_GOOGLE,
  themes,
  COLOR_BUTTON_APPLE_LIGHT,
  COLOR_BUTTON_APPLE_DARK,
  COLOR_BUTTON_APPLE_BLACK,
  COLOR_BUTTON_TEXT_APPLE_LIGHT,
  COLOR_BUTTON_TEXT_APPLE_DARK,
  COLOR_ORANGE,
  COLOR_WHITE,
  NAV_BAR_START,
  NAV_BAR_END,
} from '../../constants/colors';
import sharedStyles from '../../views/Styles';
import images from '../../assets/images';

/* eslint-disable react-native/no-unused-styles */
const styles = StyleSheet.create({
  container: {
    borderRadius: 13,
    backgroundColor: COLOR_WHITE,
  },
  oauthButtonContainer: {
    margin: 8,
  },
  oauthBtn: {
    height: 40,
    resizeMode: 'contain',
  },
  gradientButtonContainer: {
    width: '100%',
    borderRadius: 13,
  },
  gradientButtonText: {
    borderRadius: 13,
    height: 45,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 16,
  },
  buttonContainer: {
    borderRadius: 13,
    height: 48,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 16,
    borderWidth: 1.5,
  },
  oauthIcon: {
    width: 19,
    height: 19,
    marginRight: 4,
    resizeMode: 'contain',
  },
  icon: {
    position: 'absolute',
    left: 7.5,
    top: 7.5,
    width: 35,
    height: 35,
    resizeMode: 'contain',
  },
  text: {
    ...sharedStyles.textMedium,
    fontSize: 16,
    letterSpacing: 1.6,
    textAlign: 'center',
  },
  shadow: {
    shadowColor: '#000',
    shadowRadius: 2,
    shadowOpacity: 0.4,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    elevation: 8,
  },
  background_primary: {
    backgroundColor: COLOR_BUTTON_PRIMARY,
  },
  background_secondary: {
    backgroundColor: COLOR_BUTTON_SECONDARY,
  },
  background_default: {
    backgroundColor: COLOR_BUTTON_DEFAULT,
  },
  background_danger: {
    backgroundColor: COLOR_BUTTON_DANGER,
  },
  background_white: {
    backgroundColor: COLOR_BUTTON_WHITE,
    borderWidth: 1,
    borderColor: '#EBEBEB',
  },
  background_grey: {
    backgroundColor: COLOR_BUTTON_GRAY,
    borderWidth: 1,
    borderColor: '#EBEBEB',
  },
  background_done: {
    backgroundColor: COLOR_BUTTON_DONE,
  },
  background_facebook: {
    backgroundColor: COLOR_BUTTON_FACEBOOK,
  },
  background_google: {
    backgroundColor: COLOR_BUTTON_GOOGLE,
  },
  background_apple_light: {
    backgroundColor: COLOR_BUTTON_APPLE_LIGHT,
  },
  background_apple_dark: {
    backgroundColor: COLOR_BUTTON_APPLE_DARK,
  },
  background_apple_black: {
    backgroundColor: COLOR_BUTTON_APPLE_BLACK,
  },
  text_color_primary: {
    color: COLOR_BUTTON_TEXT_PRIMARY,
  },
  text_color_secondary: {
    color: COLOR_BUTTON_TEXT_SECONDARY,
  },
  text_color_default: {
    color: COLOR_BUTTON_TEXT_DEFAULT,
  },
  text_color_danger: {
    color: COLOR_BUTTON_TEXT_DANGER,
  },
  text_color_white: {
    color: COLOR_BUTTON_TEXT_WHITE,
  },
  text_color_done: {
    color: COLOR_BUTTON_TEXT_DONE,
  },
  text_color_facebook: {
    color: COLOR_BUTTON_TEXT_FACEBOOK,
  },
  text_color_google: {
    color: COLOR_BUTTON_TEXT_GOOGLE,
  },
  text_color_apple_light: {
    color: COLOR_BUTTON_TEXT_APPLE_LIGHT,
  },
  text_color_apple_dark: {
    color: COLOR_BUTTON_TEXT_APPLE_DARK,
  },

  button_size_Z: {
    width: 335,
  },
  button_size_Y: {
    width: 201,
  },
  button_size_X: {
    width: 125,
  },
  button_size_W: {
    width: '100%',
  },
  button_size_V: {
    width: 125,
  },
  button_size_U: {
    width: 140,
  },
  button_size_T: {
    width: 313,
  },
  button_size_S: {
    width: 345,
  },

  font_weight_regular: {
    ...sharedStyles.textRegular,
  },
  font_weight_medium: {
    ...sharedStyles.textMedium,
  },
  font_weight_semibold: {
    ...sharedStyles.textSemibold,
  },
  font_weight_bold: {
    ...sharedStyles.textBold,
  },
});

const Button = props => {
  const {
    title,
    text,
    icon,
    type,
    size,
    onPress,
    iconCenter,
    hidden,
    disabled,
    width,
    height,
    backgroundColor,
    loading,
    style,
    textColor: color,
    fontWeight,
    theme,
    pressingHighlight,
    ...otherProps
  } = props;

  const [isPress, setIsPress] = React.useState(false);

  if (hidden) {
    return null;
  }

  if (type === 'oauth') {
    return (
      <RectButton onPress={onPress} style={styles.oauthButtonContainer}>
        <Image source={images[title]} style={styles.oauthBtn} />
      </RectButton>
    );
  }

  if (type === 'gradient') {
    return (
      <TouchableHighlight
        onPress={onPress}
        underlayColor='white'
        onHideUnderlay={_ => setIsPress(false)}
        onShowUnderlay={_ => {if (pressingHighlight) setIsPress(true);}}
        style={[styles.gradientButtonContainer, style]}
        disabled={(disabled || loading)}>
        <LinearGradient
          colors={!isPress ? [themes[theme].bodyText, themes[theme].bodyText] : [themes[theme].backgroundColor, themes[theme].backgroundColor]}
          style={[
            styles.gradientButtonText,
            size ? styles[`button_size_${size}`] : {},
            { borderColor: themes[theme].bodyText, borderWidth: 2, backgroundColor: !isPress ? themes[theme].bodyText : themes[theme].backgroundColor }
          ]}>
          {loading ? (
            <ActivityIndicator color={'white'} />
          ) : (
            <>
              {iconCenter ? iconCenter(isPress ? { tintColor: themes[theme].bodyText } : undefined) : null}
              <Text
                style={{
                  ...styles.text,
                  color: disabled ? themes[theme].auxiliaryText : isPress ? themes[theme].bodyText : 'white',
                  fontWeight: fontWeight
                    ? styles[`font_weight_${fontWeight}`]
                    : 'normal',
                }}>
                {text || title}
              </Text>
            </>
          )}
        </LinearGradient>
      </TouchableHighlight>
    );
  }

  return (
    <TouchableHighlight
      onPress={onPress}
      underlayColor={themes[theme].bodyText}
      onHideUnderlay={_ => setIsPress(false)}
      onShowUnderlay={_ => {if (pressingHighlight) setIsPress(true);}}
      disabled={(disabled || loading)}
      style={[
        styles.container,
        size ? styles[`button_size_${size}`] : {},
        style,
      ]}
      {...otherProps}>
      <View
        style={[
          styles.buttonContainer,
          size ? styles[`button_size_${size}`] : {},
          width ? { width } : {},
          height ? { height } : {},
          { backgroundColor: isPress ? themes[theme].bodyText : 'white' }
        ]}>
        {icon ? (
          <FontAwesomeIcon
            name={icon}
            size={24}
            style={{ marginRight: 8, color: 'black' }}
          />
        ) : null}
        {loading ? (
          <ActivityIndicator color={color ?? themes[theme].buttonText} />
        ) : (
          <>
            {iconCenter ? iconCenter(isPress ? { tintColor: 'white' } : undefined) : null}
            <Text
              style={{
                ...styles.text,
                color: disabled ? themes[theme].auxiliaryText : !isPress ? themes[theme].bodyText : 'white',
                fontWeight: fontWeight
                  ? styles[`font_weight_${fontWeight}`]
                  : 'normal',
              }}>
              {text || title}
            </Text>
          </>
        )}
      </View>
    </TouchableHighlight>
  );
};

Button.propTypes = {
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  icon: PropTypes.string,
  iconCenter: PropTypes.func,
  type: PropTypes.string,
  size: PropTypes.string,
  onPress: PropTypes.func,
  hidden: PropTypes.bool,
  disabled: PropTypes.bool,
  width: PropTypes.string,
  height: PropTypes.string,
  backgroundColor: PropTypes.string,
  loading: PropTypes.bool,
  style: PropTypes.any,
  textColor: PropTypes.string,
  fontWeight: PropTypes.string,
  theme: PropTypes.string,
};

Button.defaultProps = {
  title: 'Press me!',
  type: 'primary',
  size: 'Z',
  onPress: () => alert('It works!'),
  icon: null,
  hidden: false,
  disabled: false,
  loading: false,
};

export default Button;
