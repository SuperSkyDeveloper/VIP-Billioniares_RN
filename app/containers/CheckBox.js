import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';

import Ionicons from 'react-native-vector-icons/Ionicons';
import { COLOR_BLACK_LIGHT, COLOR_BLUE } from '../constants/colors';

const CheckBox = React.memo(
  ({
    title,
    checked,
    checkedColor,
    unCheckedColor,
    checkedIcon,
    uncheckedIcon,
    textStyle,
    containerStyle,
    onPress,
  }) => (
    <TouchableOpacity
      style={{
        ...containerStyle,
        flexDirection: 'row',
        alignItems: 'center',
        height: 49,
        paddingHorizontal: 12,
        borderWidth: 1,
        borderRadius: 11,
        borderColor: checked ? COLOR_BLUE : COLOR_BLACK_LIGHT,
        marginBottom: 5,
      }}
      onPress={onPress}>
      <Ionicons
        name={checked ? checkedIcon : uncheckedIcon}
        size={12}
        color={checked ? checkedColor : unCheckedColor}
      />
      {title ? (
        <Text style={{ marginLeft: 8, ...textStyle }}>{title}</Text>
      ) : null}
    </TouchableOpacity>
  ),
);

CheckBox.propTypes = {
  title: PropTypes.string,
  checked: PropTypes.bool,
  checkedColor: PropTypes.string,
  unCheckedColor: PropTypes.string,
  checkedIcon: PropTypes.string,
  uncheckedIcon: PropTypes.string,
  textStyle: PropTypes.object,
  containerStyle: PropTypes.object,
  onPress: PropTypes.func,
};

export default CheckBox;
