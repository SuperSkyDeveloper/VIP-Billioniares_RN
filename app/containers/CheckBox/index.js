import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';

import { COLOR_YELLOW } from '../../constants/colors';
import { VectorIcon } from '../VectorIcon';

const Index = React.memo(
  ({ title, checked, textStyle, containerStyle, onPress }) => (
    <TouchableOpacity
      style={{
        ...containerStyle,
        flexDirection: 'row',
        alignItems: 'center',
        paddingVertical: 8,
      }}
      onPress={onPress}>
      <VectorIcon
        name={checked ? 'check-square-o' : 'square-o'}
        type={'FontAwesome'}
        size={24}
        color={checked ? COLOR_YELLOW : 'black'}
      />
      {title ? (
        <Text style={{ marginLeft: 8, ...textStyle }}>{title}</Text>
      ) : null}
    </TouchableOpacity>
  ),
);

Index.propTypes = {
  title: PropTypes.string,
  checked: PropTypes.bool,
  checkedColor: PropTypes.string,
  unCheckedColor: PropTypes.string,
  checkedIcon: PropTypes.string,
  uncheckedIcon: PropTypes.string,
  textStyle: PropTypes.object,
  containerStyle: PropTypes.object,
  onPress: PropTypes.func,
};

export default Index;
