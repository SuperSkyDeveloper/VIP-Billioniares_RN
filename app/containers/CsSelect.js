import React, { useState } from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import PropTypes from 'prop-types';
import RNPickerSelect from 'react-native-picker-select';

import sharedStyles from '../views/Styles';
import { themes } from '../constants/colors';
import { isIOS } from '../utils/deviceInfo';
import { VectorIcon } from './VectorIcon';

const styles = StyleSheet.create({
  container: {
    marginBottom: 10,
  },
  label: {
    paddingHorizontal: 3,
    backgroundColor: 'white',
    position: 'absolute',
    left: 10,
    zIndex: 1,
    fontSize: 11,
    top: -7,
    color: '#ffffffbb',
  },
  iosPadding: {
    height: 50,
    justifyContent: 'center',
  },
  wrap: {
    position: 'relative',
  },
  viewContainer: {
    justifyContent: 'center',
    borderRadius: 15,
    borderWidth: 1,
    flexGrow: 1,
  },
  pickerText: {
    ...sharedStyles.textRegular,
    fontSize: 14,
    paddingHorizontal: 14,
  },
  icon: {
    marginRight: 12,
  },
  iosIcon: {
    paddingVertical: 10,
  },
  loading: {
    padding: 0,
  },
  iconStyle: {
    position: 'absolute',
    right: 12,
    paddingTop: 14,
  },
});

export const CsSelect = React.memo(
  ({
    label,
    containerStyle,
    options = [],
    placeholder,
    onChange,
    disabled,
    value: initialValue,
    theme,
  }) => {
    const [selected, setSelected] = useState(
      !Array.isArray(initialValue) && (initialValue === 0 ? -1 : initialValue),
    );
    const items = options.map(option => ({
      label: option.text,
      value: option.value === 0 ? -1 : option.value,
    }));
    const pickerStyle = {
      ...styles.viewContainer,
      ...(isIOS ? styles.iosPadding : {}),
      borderColor: '#999999',
      backgroundColor: themes[theme].backgroundColor,
    };

    return (
      <View style={[styles.container, containerStyle]}>
        {label ? (
          <Text
            contentDescription={null}
            accessibilityLabel={null}
            style={[
              styles.label,
              {
                backgroundColor: themes[theme].backgroundColor,
              },
            ]}>
            {label}
          </Text>
        ) : null}
        <View style={styles.wrap}>
          <RNPickerSelect
            items={items}
            placeholder={placeholder ? { label: placeholder, value: null } : {}}
            useNativeAndroidPickerStyle={false}
            value={selected}
            disabled={disabled}
            onValueChange={value => {
              setSelected(value);
              onChange(value === -1 ? 0 : value);
            }}
            style={{
              viewContainer: pickerStyle,
              inputAndroidContainer: pickerStyle,
            }}
            textInputProps={{
              style: {
                ...styles.pickerText,
                color: selected
                  ? themes[theme].titleText
                  : themes[theme].auxiliaryText,
              },
            }}
          />
          <VectorIcon
            type={'Entypo'}
            name={!selected ? 'chevron-thin-up' : 'chevron-thin-down'}
            color={themes[theme].activeTintColor}
            size={20}
            style={styles.iconStyle}
          />
        </View>
      </View>
    );
  },
);

CsSelect.propTypes = {
  options: PropTypes.array,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
  loading: PropTypes.bool,
  disabled: PropTypes.bool,
  value: PropTypes.array,
  theme: PropTypes.string,
};
