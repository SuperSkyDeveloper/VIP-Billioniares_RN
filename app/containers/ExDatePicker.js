import React, { useState, useMemo, useRef } from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';
import DatePicker, { getFormatedDate } from 'react-native-modern-datepicker';
import { TextInput } from 'react-native-paper';

import { date_str_format, DATE_STRING_DISPLAY_FORMAT } from '../utils/datetime';
import { COLOR_BLUE, COLOR_YELLOW, themes } from '../constants/colors';
import sharedStyles from '../views/Styles';
import { VectorIcon } from './VectorIcon';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginBottom: 10,
  },
  label: {
    marginBottom: 4,
    fontSize: 14,
    ...sharedStyles.textSemibold,
  },
  content: {},
  selectContainer: {
    flex: 1,
    borderRadius: 12,
    paddingBottom: 8,
    width: '100%',
    shadowColor: 'black',
    shadowOpacity: 0.2,
    shadowOffset: { x: 2, y: 2 },
    shadowRadius: 4,
    elevation: 2,
    marginTop: 8,
  },
  valueContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 24,
    borderWidth: 1,
    height: 36,
  },
  selectContent: {
    flexDirection: 'row',
    height: 120,
  },
  headerContainer: {
    height: 36,
    position: 'relative',
    alignItems: 'center',
    justifyContent: 'center',
  },
  doneText: {
    color: 'white',
    position: 'absolute',
    right: 4,
    top: 8,
    fontWeight: 'bold',
  },
  daySelect: {
    width: 80,
  },
  monthSelect: {
    flexGrow: 1,
    marginHorizontal: 8,
  },
  yearSelect: {
    width: 80,
  },
  selectHeader: {
    textAlign: 'center',
    fontSize: 16,
  },
  selectedStyle: {
    paddingVertical: 4,
    paddingHorizontal: 8,
    borderRadius: 8,
    backgroundColor: COLOR_YELLOW,
  },
  selectStyle: {
    paddingVertical: 4,
    paddingHorizontal: 8,
  },
  selectValue: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 16,
    textAlign: 'center',
  },
  value: {
    marginRight: 12,
  },
  iconStyle: {
    position: 'absolute',
    right: 12,
  },
  textInput: {
    flex: 1,
    height: 50,
    marginBottom: 3,
  },
  iconWrap: {
    flex: 1,
    justifyContent: 'center',
  },
});

const calendarLightOption = {
  mainColor: COLOR_BLUE,
  borderColor: 'transparent',
};

const calendarDarkOption = {
  mainColor: COLOR_BLUE,
  borderColor: 'transparent',
  backgroundColor: 'transparent',
  textHeaderColor: COLOR_BLUE,
  textDefaultColor: 'white',
};

const ExDatePicker = props => {
  const [show, setShow] = useState(false);
  const { label, containerStyle, theme, error, value: currentDate } = props;
  const inputBox = useRef(null);

  const rightIcon = () => {
    return (
      <View style={styles.iconWrap}>
        <VectorIcon
          type={'Entypo'}
          name={show ? 'chevron-thin-up' : 'chevron-thin-down'}
          color={themes[theme].activeTintColor}
          size={18}
        />
      </View>
    );
  };

  const selectedDate = useMemo(() => {
    if (currentDate) {
      const units = currentDate.split('/');
      const newDate = `${units[2]}-${units[0]}-${units[1]}`;
      return newDate;
    } else {
      return getFormatedDate(new Date(), 'YYYY/MM/DD');
    }
  }, [currentDate]);

  return (
    <View style={[styles.container, containerStyle]}>
      <View style={styles.content}>
        <TextInput
          ref={ref => {inputBox.current = ref;}}
          label={label}
          onFocus={() => setShow(true)}
          onBlur={() => setShow(false)}
          value={
            currentDate
              ? date_str_format(currentDate, DATE_STRING_DISPLAY_FORMAT)
              : null
          }
          mode="outlined"
          style={[
            styles.textInput,
            {
              fontSize: Platform.OS === 'ios' ? 14 : 13,
              lineHeight: Platform.OS === 'ios' ? 14 : 14,
              backgroundColor: themes[theme].backgroundColor,
            },
          ]}
          outlineColor={error ? '#DD2E2E' : '#888888'}
          activeOutlineColor={error ? '#DD2E2E' : themes[theme].infoText}
          theme={{
            roundness: 15,
            borderWidth: 1,
            colors: {
              text: themes[theme].activeTintColor,
              placeholder: themes[theme].infoText,
            },
          }}
          right={
            <TextInput.Icon
              name={rightIcon}
              style={{ marginTop: 15 }}
              onPress={() => setShow(!show)}
            />
          }
          showSoftInputOnFocus={false}
        />
        {show ? (
          <View style={styles.selectContainer}>
            <DatePicker
              mode="calendar"
              style={[
                { borderRadius: 10 },
                theme === 'dark' && { borderWidth: 1, borderColor: 'white' },
              ]}
              options={theme === 'dark' ? calendarDarkOption : calendarLightOption}
              current={selectedDate}
              selected={selectedDate}
              onSelectedChange={date => {
                console.log('MMM - ' + date);
                let units = date.split('/');
                const newDate = `${units[1]}/${units[2]}/${units[0]}`;
                units = selectedDate.split(/-|\//);
                const selDate = `${units[1]}/${units[2]}/${units[0]}`;
                if (newDate !== selDate) {
                  if (inputBox.current) {
                    inputBox.current.blur();
                  }
                  props.action({ value: newDate });
                }
              }}
            />
          </View>
        ) : null}
      </View>
    </View>
  );
};

ExDatePicker.PropTypes = {
  label: PropTypes.string,
  containerStyle: PropTypes.object,
  value: PropTypes.string,
  action: PropTypes.func,
  toggleShow: PropTypes.func,
  topScrollEnable: PropTypes.bool,
  theme: PropTypes.string,
};

export default ExDatePicker;
