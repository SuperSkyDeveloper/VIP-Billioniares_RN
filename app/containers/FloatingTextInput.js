import React, { useState } from 'react';
import { View, StyleSheet, Image, Text, Platform } from 'react-native';
import PropTypes from 'prop-types';
import { TextInput } from 'react-native-paper';

import { themes } from '../constants/colors';
import images from '../assets/images';

const styles = StyleSheet.create({
  iconWrap: {
    flex: 1,
    justifyContent: 'center',
  },
  icon: {
    width: 20,
    height: 20,
    resizeMode: 'contain',
  },
  textInput: {
    flex: 1,
    height: 50,
    marginBottom: 3,
  },
  error: {
    fontSize: 9,
    fontWeight: '300',
    color: '#DD2E2E',
  },
  container: {
    marginBottom: 14,
  },
});

const FloatingTextInput = props => {
  const {
    label,
    required,
    error,
    loading,
    secureTextEntry,
    containerStyle,
    inputRef,
    iconLeft,
    iconRight,
    inputStyle,
    wrapStyle,
    testID,
    placeholder,
    theme,
    outlineColor,
    backgroundColor,
    multiline,
    value,
    ...inputProps
  } = props;

  const [showPassword, setShowPassword] = useState(!secureTextEntry);

  const { dangerColor } = themes[theme];

  const leftIcon = () => {
    return (
      <View style={styles.iconWrap}>
        <Image
          source={iconLeft}
          testID={testID ? `${testID}-icon-left` : null}
          style={[styles.icon, { tintColor: themes[theme].infoText }]}
        />
      </View>
    );
  };

  const rightIcon = () => {
    if (error) {
      return (
        <View style={styles.iconWrap}>
          <Image
            source={images.error}
            testID={testID ? `${testID}-icon-error` : null}
            style={[styles.icon]}
          />
        </View>
      );
    }
    if (secureTextEntry) {
      return (
        <View style={styles.iconWrap}>
          <Image
            source={showPassword ? images.eye_show : images.eye_hidden}
            testID={testID ? `${testID}-icon-right` : null}
            style={[styles.icon, { tintColor: themes[theme].infoText }]}
          />
        </View>
      );
    }
  };

  const tooglePassword = () => {
    setShowPassword(!showPassword);
  };

  return (
    <View style={styles.container}>
      <TextInput
        ref={inputRef}
        value={value}
        label={label}
        mode="outlined"
        style={[
          styles.textInput,
          {
            backgroundColor: backgroundColor ?? themes[theme].backgroundColor,
            height: multiline ? 123 : 50,
            fontSize: Platform.OS === 'ios' ? 14 : 13,
            lineHeight: Platform.OS === 'ios' ? 14 : 14,
          },
        ]}
        outlineColor={error ? '#DD2E2E' : outlineColor || '#888888'}
        activeOutlineColor={error ? '#DD2E2E' : themes[theme].infoText}
        theme={{
          roundness: 15,
          borderWidth: 1,
          colors: {
            text: themes[theme].activeTintColor,
            placeholder: themes[theme].infoText,
          },
        }}
        left={
          iconLeft && <TextInput.Icon name={leftIcon} style={{ marginTop: 15 }} />
        }
        right={
          (error || secureTextEntry) && (
            <TextInput.Icon
              name={rightIcon}
              style={{ marginTop: 15 }}
              onPress={tooglePassword}
            />
          )
        }
        multiline={multiline}
        secureTextEntry={!showPassword}
        {...inputProps}
      />
      {error ? <Text style={styles.error}>{error}</Text> : null}
    </View>
  );
};

FloatingTextInput.propTypes = {
  label: PropTypes.string,
  required: PropTypes.string,
  error: PropTypes.string,
  loading: PropTypes.bool,
  secureTextEntry: PropTypes.bool,
  containerStyle: PropTypes.object,
  inputStyle: PropTypes.object,
  inputRef: PropTypes.func,
  testID: PropTypes.string,
  iconLeft: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  iconRight: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  placeholder: PropTypes.string,
  onIconRightPress: PropTypes.func,
  theme: PropTypes.string,
  outlineColor: PropTypes.string,
  backgroundColor: PropTypes.string,
  multiline: PropTypes.bool,
};

FloatingTextInput.defaultProps = {
  error: '',
};

export default FloatingTextInput;
