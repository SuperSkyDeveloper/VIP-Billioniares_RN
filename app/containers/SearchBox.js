import React from 'react';
import { View, StyleSheet, TextInput } from 'react-native';
import PropTypes from 'prop-types';

import sharedStyles from '../views/Styles';
import { withTheme } from '../theme';
import { themes, COLOR_BLACK } from '../constants/colors';
import { VectorIcon } from './VectorIcon';

const styles = StyleSheet.create({
  container: {},
  searchBox: {
    alignItems: 'center',
    borderRadius: 24,
    flexDirection: 'row',
    fontSize: 17,
    paddingHorizontal: 10,
    marginVertical: 4,
    marginHorizontal: 8,
  },
  input: {
    height: 40,
    flex: 1,
    fontSize: 17,
    marginLeft: 8,
    paddingTop: 0,
    paddingBottom: 0,
    ...sharedStyles.textRegular,
  },
  cancel: {
    padding: 4,
  },
  cancelText: {
    ...sharedStyles.textRegular,
    fontSize: 17,
  },
});

export const SearchBox = ({
  onChangeText,
  onSubmitEditing,
  testID,
  hasCancel,
  onCancelPress,
  inputRef,
  placeholder,
  theme,
  ...props
}) => (
  <View
    style={[
      styles.container,
      { backgroundColor: themes[theme].backgroundColor },
    ]}>
    <View style={[styles.searchBox, { backgroundColor: themes[theme].searchboxBackground }]}>
      <VectorIcon
        name="search"
        type="Ionicons"
        size={18}
        color={themes[theme].auxiliaryText}
      />
      <TextInput
        ref={inputRef}
        autoCapitalize="none"
        autoCorrect={false}
        blurOnSubmit
        clearButtonMode="while-editing"
        placeholder={placeholder ?? 'Search'}
        returnKeyType="search"
        style={[styles.input, { color: themes[theme].activeTintColor }]}
        testID={testID}
        underlineColorAndroid="transparent"
        onChangeText={onChangeText}
        onSubmitEditing={onSubmitEditing}
        placeholderTextColor={themes[theme].auxiliaryText}
        theme={theme}
        {...props}
      />
    </View>
  </View>
);

SearchBox.propTypes = {
  onChangeText: PropTypes.func.isRequired,
  onSubmitEditing: PropTypes.func,
  hasCancel: PropTypes.bool,
  onCancelPress: PropTypes.func,
  theme: PropTypes.string,
  inputRef: PropTypes.func,
  testID: PropTypes.string,
  placeholder: PropTypes.string,
};

export default withTheme(SearchBox);
