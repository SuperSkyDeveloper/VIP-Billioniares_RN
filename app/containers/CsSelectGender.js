import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import PropTypes from 'prop-types';

import sharedStyles from '../views/Styles';
import {
  COLOR_DANGER,
  COLOR_BLUE,
  themes,
} from '../constants/colors';
import CheckBox from './CheckBox';
import { Genders } from '../constants/app';

const styles = StyleSheet.create({
  inputContainer: {
    marginBottom: 10,
  },
  label: {
    marginBottom: 4,
    fontSize: 14,
    ...sharedStyles.textSemibold,
  },
  required: {
    marginBottom: 10,
    color: COLOR_DANGER,
    fontSize: 14,
    fontWeight: '700',
  },
  selectText: {
    ...sharedStyles.textRegular,
    fontSize: 14,
  },
  wrap: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    flexWrap: 'wrap',
  },
  CheckBoxContainer: {},
});

const CsSelectGender = (props) => {
  const { 
    label, 
    value, 
    required, 
    containerStyle, 
    theme 
  } = props;
  
  const setCheck = value => {
    props.onChange(value);
  };

  return (
    <View style={[styles.inputContainer, containerStyle]}>
      {label ? (
        <Text
          contentDescription={null}
          accessibilityLabel={null}
          style={[styles.label, { color: themes[theme].titleText }]}>
          {label}
          {required ? (
            <Text
              contentDescription={null}
              accessibilityLabel={null}
              style={[styles.required]}>{` ${required}`}</Text>
          ) : null}
        </Text>
      ) : null}
      <View
        style={styles.wrap}>
        {Genders.map(val => (
          <CheckBox
            key={`gender-key-${val.value}`}
            title={val.text}
            checked={value === val.value}
            onPress={() => setCheck(val.value)}
            onIconPress={() => setCheck(val.value)}
            checkedIcon="radio-button-on"
            uncheckedIcon="radio-button-off"
            checkedColor={COLOR_BLUE}
            unCheckedColor={themes[theme].bodyText}
            textStyle={{ color: themes[theme].bodyText }}
            containerStyle={{ ...styles.CheckBoxContainer }}
          />
        ))}
      </View>
    </View>
  );
};

CsSelectGender.PropTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  required: PropTypes.string,
  containerStyle: PropTypes.object,
  onChange: PropTypes.func,
  theme: PropTypes.string,
};

export default CsSelectGender;