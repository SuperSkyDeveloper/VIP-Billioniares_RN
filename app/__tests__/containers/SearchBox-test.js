import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react-native';
import { SearchBox } from '../../containers/SearchBox';

test('renders correctly', () => {
  const tree = renderer.create(<SearchBox onChangeText={() => {}} theme="dark"/>).toJSON();
  expect(tree).toMatchSnapshot();
});