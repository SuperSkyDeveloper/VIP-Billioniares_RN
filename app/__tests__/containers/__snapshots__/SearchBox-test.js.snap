// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "backgroundColor": "#080808",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 24,
          "flexDirection": "row",
          "fontSize": 17,
          "marginHorizontal": 8,
          "marginVertical": 4,
          "paddingHorizontal": 10,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#9ca2a8",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": undefined,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï”…
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={true}
      clearButtonMode="while-editing"
      onChangeText={[Function]}
      placeholder="Search"
      placeholderTextColor="#9ca2a8"
      rejectResponderTermination={true}
      returnKeyType="search"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#000000",
          "flex": 1,
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "400",
          "height": 40,
          "marginLeft": 8,
          "paddingBottom": 0,
          "paddingTop": 0,
        }
      }
      theme="dark"
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
